### PR TITLE
CPP: Tiny qldoc fixes.

### PR DESCRIPTION
**Pull request by geoffw0**
Lots of tiny QLDoc corrections / cleanup, many of which were accumulated while reviewing large quantities of code last week.

### CODE REVIEW

**Comment by zlaski-semmle** [comment]
Please remove `Type.qll` from your list, as I have one with very similar changes sitting on the `zlaski/cpp418` branch.  Other than that, LGTM.

File: cpp/ql/src/semmle/code/cpp/Type.qll [hunk 1]
```diff
@@ -1001,7 +1003,7 @@ class ArrayType extends DerivedType {
  * A GNU/Clang vector type.
  *
  * In both Clang and GNU compilers, vector types can be introduced using the
- * __attribute__((vector_size(byte_size))) syntax. The Clang compiler also
+ * `__attribute__((vector_size(byte_size)))` syntax. The Clang compiler also
  * allows vector types to be introduced using the ext_vector_type,
  * neon_vector_type, and neon_polyvector_type attributes (all of which take
  * an element type rather than a byte size).
```
Comment (zlaski-semmle) [inline cpp/ql/src/semmle/code/cpp/Type.qll]: Also back-tick `ext_vector_type`, `neon_vector_type` and `neon_polyvector_type`.

File: cpp/ql/src/semmle/code/cpp/Type.qll [hunk 2]
```diff
@@ -1243,7 +1245,12 @@ class TemplateParameter extends UserType {
   override predicate involvesTemplateParameter() { any() }
 }
 
-/** A C++ template template parameter, e.g. template &lt;template &lt;typename,typename> class T>. */
+/**
+ * A C++ template template parameter, for example:
+ * ```
+ * template <template <typename,typename> class T>
+ * ```
+ */
```
Comment (zlaski-semmle) [inline cpp/ql/src/semmle/code/cpp/Type.qll]: I wonder if we should not use an example that motivates the use of template template parameters.  Here's what I found on the internet (and performed renaming):
```
template <template<typename> class Container, class Element>
void First(const Container<Element> &container) { }
```
What do you think?

**Comment by geoffw0** [comment]
> Please remove Type.qll from your list

Would you prefer me to undo all the changes to `Type.qll` here or finish the changes you've recommended?

**Comment by zlaski-semmle** [comment]
> > Please remove Type.qll from your list
> 
> Would you prefer me to undo all the changes to `Type.qll` here or finish the changes you've recommended?
The former.  Just discard the changes.   We somehow both wound up editing the same file.

**Comment by geoffw0** [comment]
Reverted changes to Type.qll.

**Comment by geoffw0** [comment]
Is there anything stopping this from being merged?

**Pull request merged**